src/main/scala/SerialAdapter.scala

```diff
@@ -32,6 +38,29 @@ case object SerialAdapter {
     ram
   }
 
+  def connectHarnessMultiClockAXIRAM(serdesser: TLSerdesser, serial_port: ClockedIO[SerialIO], mem_clock_port: ClockBundle, reset: Reset): MultiClockSerialAXIRAM = {
+    implicit val p: Parameters = serdesser.p
+
+    val ram = LazyModule(new MultiClockSerialAXIRAM(
+      p(SerialTLKey).get.width,
+      p(SerialTLKey).get.memParams,
+      p(SerialTLKey).get.axiMemOverSerialTLParams.get,
+      managerEdge = serdesser.managerNode.edges.in(0),
+      clientEdge = serdesser.clientNode.edges.out(0)
+    ))
+
+    withClockAndReset(serial_port.clock, reset) {
+      val module = Module(ram.module)
```

Comment:
Can you look at my changes in https://github.com/ucb-bar/testchipip/pull/124? 
I think its better to clock this module by the harnessClock, and have AsyncQueues do the crossing, rather than giving this module the serial clock directly.

Using AsyncQueues is also safer for a chip with a more realistic clock/reset setup, since there is no guarantee on synchronization between harness-level reset and the reset for the on-chip serdes.

Reply:
Sure. I think the cleanest way to integrate those changes is to merge that in Chipyard 1st then this PR will get updated with those changes.